### ISSUE TITLE
IMPLICIT NONE is not inherited by nested scope

### ISSUE THREAD

**Issue by tskeith** [flang-compiler/f18#71]
In the example below, the compiler should report an error on the use of `x`:
```
program test
  implicit none
contains
  subroutine foo
    x = 666 ! wrong since x has no implicit rule
  end subroutine
end program
```
